### ISSUE TITLE
IncompleteJSONError: lexical error: invalid char in json text.

### ISSUE THREAD

**Issue by Adaickalavan** [huawei-noah/SMARTS#1453]
## **BUG REPORT**
**High Level Description**

An `IncompleteJSONError('lexical error: invalid char in json text.` error is thrown by `envision/server.py` while running smarts with head.

Error attached is below.

```
ERROR:tornado.application:Exception in callback functools.partial(<function WebSocketProtocol._run_callback.<locals>.<lambda> at 0x7f28a0b5b200>, <Task finished coro=<BroadcastWebSocket.on_message() done, defined at /home/kyber/workspace/SMARTS/examples/rl/intersection/.venv/lib/python3.7/site-packages/envision/server.py:309> exception=IncompleteJSONError('lexical error: invalid char in json text.\n          1.84, -1.6, -71.35, -1.56], [Infinity, Infinity, Infinity, I\n                     (right here) ------^\n')>)
Traceback (most recent call last):
  File "/home/kyber/workspace/SMARTS/examples/rl/intersection/.venv/lib/python3.7/site-packages/tornado/ioloop.py", line 741, in _run_callback
    ret = callback()
  File "/home/kyber/workspace/SMARTS/examples/rl/intersection/.venv/lib/python3.7/site-packages/tornado/websocket.py", line 656, in <lambda>
    self.stream.io_loop.add_future(result, lambda f: f.result())
  File "/home/kyber/workspace/SMARTS/examples/rl/intersection/.venv/lib/python3.7/site-packages/envision/server.py", line 314, in on_message
    for prefix, event, value in it:
ijson.common.IncompleteJSONError: lexical error: invalid char in json text.
          1.84, -1.6, -71.35, -1.56], [Infinity, Infinity, Infinity, I
                     (right here) ------^
```

**SMARTS version**
version: 0.6

**Previous associated issues**
None

**Steps to reproduce the bug**
In first terminal:
```
$ git checkout develop
$ cd SMARTS/examples/rl/intersection
$ python3.7 -m venv ./.venv
$ source ./.venv/bin/activate
$ pip install --upgrade pip
$ pip install -e .
$ scl envision start -s ./.venv/lib/python3.7/site-packages/smarts/scenarios/
```
In second terminal:
```
$ cd SMARTS/examples/rl/intersection
$ source ./.venv/bin/activate
$ python3.7 run.py --head
```

**Resulting and expected behaviour**
Envision should play without throwing error.

**Error logs and screenshots**
![Error](https://user-images.githubusercontent.com/24203326/171211878-8a2732ba-4c43-4d4a-861b-275de5915be7.JPG)

**System information**
- Operating system & Environment [Please note if using Docker]: Ubuntu 18.04
- Python version [Please note if using Jupyter]: python 3.7
- SUMO version: 1.10.0
- Date when bug occurred: 31 May 2022
 
**Impact [If known]**
Unable to visualize SMARTS simulation in Envision.

**Comment by Gamenot** [comment]
Context is that this appears to be an issue with the `JSONEncoder` not getting called to encode the values appropriately.

**Comment by Gamenot** [comment]
Branch is: [intersection-v0](https://github.com/huawei-noah/SMARTS/tree/intersection-v0)

**Comment by Gamenot** [comment]
The problem occurs here based on the following lines:

https://github.com/huawei-noah/SMARTS/blob/develop/envision/server.py#L314-L320

The issue is around the json being serialized with literals. Then when it is deserialized it looks like literals like `True` are missing string wrapping.

**Comment by Gamenot** [comment]
This was fixed with #1522.
